### DISCUSSION TITLE
Is there a way to get the URL of a file from its input path?

### DISCUSSION THREAD

**Discussion by architchandra** [11ty/eleventy#2133] [Q&A] [answered]
I'm using the [Select field](https://forestry.io/docs/settings/fields/select/#document-reference-source) in Forestry to allow authors to select the pages that they want to add to a header navigation.

In this, Forestry sets the input path of the selected page (eg. `src/about.md`) in a data file. I need to find a way to get the corresponding URL (eg. `/about`) that can be used in the header navigation.

Am I missing something obvious?

**Comment by cfjedimaster** [comment]
There's a default for permalinks (https://www.11ty.dev/docs/permalinks/) where something.ext maps to /something/index.html. Individual templates can override this though.

**Comment by pdehaan** [comment, marked as answer]
A bit awkward, but this custom shortcode might work:

```js
module.exports = function (eleventyConfig) {
  eleventyConfig.addShortcode("perma_from_path", function (inputPath="", coll=[]) {
    return coll.find(item => item.inputPath === inputPath)?.url;
  });

  return {
    dir: {
      input: "src",
      output: "www",
    }
  };
};
```

### USAGE
```md
{% perma_from_path "./src/about.md", collections.all %}
```

### OUTPUT
```html
/about/
```

Basically just loops through `collections.all` to find a Template with a matching `inputPath` (and note that it seems like the `inputPath` always starts with a `./<inputDir>/`, since it always seemed to be failing if I tried doing an exact match on "src/about.md"); which is why I had to use `./src/about.md` and not just `src/about.md`. Although I'm sure we could add that logic into the shortcode if that's easier.

**Reply by pdehaan** [reply]
And I think we have to pass the `collections.all` (or any custom collection you fancy) to the custom shortcode because shortcodes don't seem to have access to the collection data (per https://github.com/11ty/eleventy/issues/813) as far as I understand.

**Comment by architchandra** [comment]
It took me some time to get back to this, but your answer really helped me, @pdehaan. Thank you! I ended up creating the following filter. (Will improve it later to make it more robust.)

```
module.exports = function pathToEntry(path, entriesList=[]) {
  return entriesList.find(entry => path === entry.inputPath.substring(2));
};
```

This can be used like `{% set blockEntry = block.target|pathToEntry(collections.all) %}`.
